### PR TITLE
Update Angular to 14

### LINKED ISSUE #255: Provide an Ivy distribution
After migrating to Angular 13, now I get this message when building:
```
⠙ Generating browser application bundles (phase: setup)...Processing legacy "View Engine" libraries:
- @auth0/auth0-angular [es2015/esm2015] (git+https://github.com/auth0/auth0-angular.git)
- @auth0/angular-jwt [es2015/esm2015] (git+https://github.com/auth0/angular2-jwt.git)
Encourage the library authors to publish an Ivy distribution.
```

I think it's related to the fact that Angular are deprecating the View Engine in favour of Ivy which means in Angular v14 the library won't be able to build any more if this change is not being addressed.

### PR DESCRIPTION
**Pull request by frederikprijck**
### Description

Updates Angular to 14. Also updates `typedoc` because of a conflicting `peerDependency` on `typescript`.
This also migrates to ESLint, as TSLint has been removed in the process of upgrading to 14.

### References
Closes #255

### Checklist

- [x] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used, if not `master`

### CODE REVIEW

**Review by stevehobbsdev — APPROVED** [review]

**Comment by mcarey1590** [comment]
Is there an ETA on when this version will be released?

**Comment by frederikprijck** [comment]
I had a conversation with the Angular team and concluded the following:

When you compile an SDK with Angular 14 and compilationMode set to partial, the SDK will only work with Angular 14+.
However, as we still need to support Angular 12 and 13, it appears to be that we have two options:

- Create a separate version for every major version of Angular
- Compile using the lowest supported Angular version, currently being 12.

Given we have no specific features from angular apart from a service and an interceptor, I believe we should be fine to stick with Angular 12, that would have full support for Angular 12, 13, 14 and 15.

However, that would mean we can not merge this PR yet, but can only do so once Angular has dropped support for Angular 12 and 13. Maintaining multiple versions comes with a cost that is not worth it for our SDK.

**Pull request closed without merging**
